packages/components/utils/src/getStringMatch/getStringMatch.ts

```diff
@@ -16,8 +16,12 @@ export interface MatchObj {
  */
 export function getStringMatch(base: string, match: string): MatchObj {
   const matchResult = { before: '', match: '', after: '' };
+  const escapedMatch = match.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
 
-  const regex = new RegExp(`(?<before>.*?)(?<match>${match})(?<after>.*)`, 'i');
+  const regex = new RegExp(
+    `(?<before>.*?)(?<match>${escapedMatch})(?<after>.*)`,
+    'i',
+  );
```

Comment:
yeah, but that would be a breaking change, since we currently return the 3 groups (before, match, after), with that approach we would only return an array with the matches.